library/core/src/num/mod.rs

```diff
@@ -1110,28 +1110,18 @@ pub fn can_not_overflow<T>(radix: u32, is_signed_ty: bool, digits: &[u8]) -> boo
     radix <= 16 && digits.len() <= mem::size_of::<T>() * 2 - is_signed_ty as usize
 }
 
-fn from_str_radix<T: FromStrRadixHelper>(src: &str, radix: u32) -> Result<T, ParseIntError> {
+fn from_ascii_radix<T: FromAsciiRadixHelper>(src: &[u8], radix: u32) -> Result<T, ParseIntError> {
     use self::IntErrorKind::*;
     use self::ParseIntError as PIE;
 
-    assert!(
-        (2..=36).contains(&radix),
-        "from_str_radix_int: must lie in the range `[2, 36]` - found {}",
-        radix
-    );
+    assert!((2..=36).contains(&radix), "radix must lie in the range `[2, 36]` - found {}", radix);
```

Comment:
All of these functions should probably have `#[track_caller]`, so that this panic points at the user code instead of the standard library.